v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -35,7 +35,10 @@ import { getGlobalHelpString } from "./helpers/getGlobalHelpString.js";
 import { getHelpString } from "./helpers/getHelpString.js";
 import { printVersionMessage } from "./version.js";
 
-export async function main(cliArguments: string[], print = console.log) {
+export async function main(
+  cliArguments: string[],
+  print: (message?: any, ...optionalParams: any[]) => void = console.log,
```

Comment:
I'll also need a similar typing for `print` in the error handler. Can we create a type for it?

Reply:
I think typing it as `(message:string) => void` if possible would be better.

Reply:
Addressed in 01c2e6f0b